### PR TITLE
Chromedriver

### PR DESCRIPTION
**Pull request by cfitz**
Previous PR to add chrome changed some of the gems used in the application, which has caused a big problem with the application. This tries to lock down the gems.

**The most likely cause:**
There a couple of dependencies that are loosely set to version which can
cause problems. rails-html-sanitizer require loofah gem ~> 2.0, but the
2.1.0 release introduced the crass gem, which might be causing a problem
with a space. Locking loofah to 2.0.3 removes crass.

**Another possible causes:**
Also there are known issues with aspace and the i18n gem > v0.7.0. This locks all apps to 0.7.0.
Also locks down all apps to use the same version of json.
Lastly removes a duplicate line in the staff UI's config/application.rb that added some Rack middleware twice.

### CODE REVIEW

**Comment by cfitz** [comment]
I was able to duplicate the problem with the previous PR by using the demo_db and doing some basic stress tests. After a few minutes, the staff UI would just return 500 with error messages of "ActionView::Template::Error (No default implementation found ):" and pointing to la line that has " ink_to strip_tags(crumb[0].to_s) , crumb[1]" . This line is for generating breadcrumbs, which is sanitizing the HTML, which uses rails-html-sanitizer, which uses loofah.

I've run serveral stress type tests on this version and it seems to not run into the same issue.

File: frontend/config/application.rb [hunk 1]
```diff
@@ -42,9 +42,6 @@ def self.extend_aspace_routes(routes_file)
     # https://github.com/rack/rack/issues/1075
     config.middleware.insert_before Rack::Runtime, MultipartBufferSize
 
-    # https://github.com/rack/rack/issues/1075
-    config.middleware.insert_before Rack::Runtime, MultipartBufferSize
```
Comment (lmcglohon) [inline frontend/config/application.rb]: @cfitz I had to add this back in. It was causing an error (see below). It couldn't compute the holdings information and when I pulled up the locations browse page, I got the attached issue - spinners in the repo holdings column.
screen shot 2017-10-17 at 9 05 38 am


I, [2017-10-17T08:51:34.205953 #1024]  INFO -- : Started GET "/search.json?filter_term=%7B%22location_uris%22%3A%22%2Flocations%2F384%22%7D&listing_only=true&type=top_container" for 0:0:0:0:0:0:0:1 at 2017-10-17 08:51:34 -0400
I, [2017-10-17T08:51:34.205970 #1024]  INFO -- : Started GET "/search.json?filter_term=%7B%22location_uris%22%3A%22%2Flocations%2F336%22%7D&listing_only=true&type=top_container" for 0:0:0:0:0:0:0:1 at 2017-10-17 08:51:34 -0400
F, [2017-10-17T08:51:34.225653 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.226047 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.228752 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.228897 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.229012 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.229111 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.229010 #1024] FATAL -- : config/application.rb:30:in `call'
F, [2017-10-17T08:51:34.229191 #1024] FATAL -- : config/application.rb:30:in `call'
F, [2017-10-17T08:51:34.229771 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.231195 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.231279 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.231185 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.231337 #1024] FATAL -- : config/application.rb:30:in `call'
F, [2017-10-17T08:51:34.233221 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.233335 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.233419 #1024] FATAL -- : config/application.rb:30:in `call'
F, [2017-10-17T08:51:34.233336 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.235371 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.235486 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.235559 #1024] FATAL -- : config/application.rb:30:in `call'
F, [2017-10-17T08:51:34.235374 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.237702 #1024] FATAL -- : NoMethodError (undefined method `+' for nil:NilClass):
F, [2017-10-17T08:51:34.237830 #1024] FATAL -- :   
F, [2017-10-17T08:51:34.237937 #1024] FATAL -- : config/application.rb:30:in `call'

Comment (lmcglohon) [reply frontend/config/application.rb]: @cfitz I am going to try this as a fix - https://github.com/rhinobird-io/activity-plugin/commit/b4d7efbea89515d8e2b667501e61fcfbfc2c4411. Will let you know how that goes.

Comment (lmcglohon) [reply frontend/config/application.rb]: @cfitz rhinobird-io/activity-plugin@b4d7efb worked. Can you do that in this PR? Then I think it is ready to go...

**Comment by cfitz** [comment]
@lmcglohon Tried again this morning with this.  Since the last PR was reverted, some of the code fell out of this PR. I put the chromedriver stuff back in, tweaked some tests to get passing as consistently as possible. Made on very small CSS tweak ( selenium was missing the button ). Uh...locked down the gems to stable versions...i think that's everything. Travis passed these on my repo, although sometimes it looks like there's network issues with them connecting to rubygems..we'll see...

This also included https://github.com/archivesspace/archivesspace/pull/1019 ( which just fixed a couple of tests in the indexer ) 

What do you think?

**Comment by lmcglohon** [comment]
@cfitz This is a lot of changes in one PR. May take me a few days to review.

**Comment by cfitz** [comment]
@lmcglohon sure thing. The bulk of the changes are in the selenium folder, but there are some changes other places...mainly tried to lock down the different gem versions. Let me know if there's anything that looks weird.

**Comment by cfitz** [comment]
@lmcglohon btw the tests all pass now on Travis with the tweaks to make the headless-tests pass: 
https://travis-ci.org/cfitz/archivesspace/builds/301663392?utm_source=github_status&utm_medium=notification 
Is there anything thing that you think I need to change?

**Comment by lmcglohon** [comment]
@cfitz Reviewing this for inclusion in release v2.2.2

**Comment by cfitz** [comment]
@lmcglohon hi, just curious about this, if there are any issues? let me know if there's anything I can do..

**Comment by lmcglohon** [comment]
@cfitz Hey Chris, I think everything is fine. Hoping to get this into master over the weekend.

**Pull request merged**